ui/components/app/set-approval-for-all-warning/set-approval-for-all-warning.js

```diff
@@ -0,0 +1,124 @@
+
+const SetApproveForAllWarning = ({
+  collectionName,
+  senderAddress,
+  name,
+  total,
+  onSubmit,
+  onCancel,
+  showWarningModal,
+}) => {
+  const t = useI18nContext();
+
+  const footer = (
+    <Box
+      display={DISPLAY.FLEX}
+      flexDirection={FLEX_DIRECTION.COLUMN}
+      justifyContent={JUSTIFY_CONTENT.SPACE_BETWEEN}
+      className="set-approval-for-all-warning__footer"
+    >
+      <Button
+        className="set-approval-for-all-warning__footer__approve-button"
+        type="danger-primary"
+        onClick={onSubmit}
+      >
+        {t('approveButtonText')}
+      </Button>
+      <Button
+        className="set-approval-for-all-warning__footer__cancel-button"
+        type="secondary"
+        onClick={onCancel}
+      >
+        {t('reject')}
+      </Button>
+    </Box>
+  );
+
+  return (
+    <Popover
+      className="set-approval-for-all-warning__content"
+      footer={footer}
+      open={showWarningModal}
+    >
+      <Box
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.ROW}
+        padding={4}
+        className="set-approval-for-all-warning__content__header"
+      >
+        <i className="fa fa-exclamation-triangle set-approval-for-all-warning__content__header__warning-icon" />
+        <Typography variant={TYPOGRAPHY.H4} fontWeight={FONT_WEIGHT.BOLD}>
+          {t('yourNFTmayBeAtRisk')}
+        </Typography>
+      </Box>
+      <Box
+        display={DISPLAY.FLEX}
+        padding={4}
+        justifyContent={JUSTIFY_CONTENT.SPACE_BETWEEN}
+        className="set-approval-for-all-warning__content__account"
+      >
+        <Box display={DISPLAY.FLEX}>
+          <Identicon address={senderAddress} diameter={32} />
+          <Typography
+            variant={TYPOGRAPHY.H5}
+            marginLeft={2}
+            className="set-approval-for-all-warning__content__account-name"
+          >
+            <b>{name}</b> {` (${shortenAddress(senderAddress)})`}
+          </Typography>
+        </Box>
+        <Typography>{`${t('total')}: ${total}`}</Typography>
+      </Box>
+
+      <Typography
+        color={COLORS.TEXT_ALTERNATIVE}
+        margin={4}
+        marginTop={4}
+        marginBottom={4}
+        variant={TYPOGRAPHY.H6}
+      >
+        {t('nftWarningContent', [
+          <strong
+            key="non_custodial_bold"
+            className="set-approval-for-all-warning__content__bold"
+          >
+            {t('nftWarningContentBold', [collectionName])}
+          </strong>,
+          <strong key="non_custodial_grey">
+            {t('nftWarningContentGrey')}
+          </strong>,
+        ])}
+      </Typography>
+    </Popover>
+  );
+};
+
+SetApproveForAllWarning.propTypes = {
+  collectionName: PropTypes.string,
+  senderAddress: PropTypes.string,
+  name: PropTypes.string,
+  total: PropTypes.string,
+  onSubmit: PropTypes.func,
+  onCancel: PropTypes.func,
+  showWarningModal: PropTypes.bool,
```

Comment:
Would be nice to get some propType descriptions here

Reply:
done